components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/OIDCClaimUtil.java

```diff
@@ -692,4 +715,531 @@ private static Map<String, Object> getUserClaimsInOidcDialect(Map<String, String
 
         return userClaimsInOidcDialect;
     }
+
+    /**
+     * This method retrieves the user attributes cached against the access token or the authorization code.
+     * Currently, this is supported for the code grant and the refresh grant.
+     *
+     * @param requestMsgCtx The context of the OAuth token request containing necessary properties.
+     * @return A map of cached user attributes against the code or the access token.
+     * @throws OAuthSystemException    If there is an error while generating the access token hash.
+     * @throws IdentityOAuth2Exception If an error occurs while selecting the OAuth2 token issuer.
+     */
+    public static Map<ClaimMapping, String> getCachedUserAttributes(OAuthTokenReqMessageContext requestMsgCtx)
+            throws OAuthSystemException, IdentityOAuth2Exception {
+
+        Map<ClaimMapping, String> userAttributes = getUserAttributesCachedAgainstAuthorizationCode(
+                getAuthorizationCode(requestMsgCtx));
+        if (log.isDebugEnabled()) {
+            log.debug("Retrieving claims cached against authorization_code for user: " +
+                    requestMsgCtx.getAuthorizedUser());
+        }
+        if (isEmpty(userAttributes)) {
```

Comment:
line 749 also checking same condition right? this is getting repeated in this method please check

Reply:
here if userAttributes are empty we trying to get userAttributes 742, then after we check empty